### PR TITLE
developing for AdamW

### PR DESCRIPTION
**Pull request by wjn1996**
## Developing AdamW optimizer for EasyNLP.
- Merging the AdamW and corresponding scheduler into the original file.
- Changing some codes in Trainer file.
- Add a new argument: ``optimizer_type``.

## Update some typos in benchmark/clue.
- adding ``get_pretrain_model_path`` in main / main_evaluate / main_predict.

### CODE REVIEW

File: benchmarks/clue/run_clue.sh [hunk 1]
```diff
@@ -30,12 +30,13 @@ if [ "$mode" = "train" ]; then
     --label_name=label \
     --label_enumerate_values=0,1 \
     --checkpoint_dir=./tmp/benchmarks/clue/$task \
+    --optimizer_type=AdamW \
```
Comment (minghui) [inline benchmarks/clue/run_clue.sh:33]: change it to:     --optimizer=AdamW \

Comment (minghui) [reply benchmarks/clue/run_clue.sh:33]: split it to two shell scripts:
run_clue_adamw.sh
run_clue_bertadam.sh

Comment (wjn1996) [reply benchmarks/clue/run_clue.sh:33]: done

File: easynlp/core/optimizers.py [hunk 2]
```diff
@@ -356,6 +376,99 @@ def step(self, closure=None):
...
+        return loss
+
+
+    def get_current_lr(self, scheduler: Union[EasyNLPWarmupLinearSchedule, None] = None):
+        if scheduler:
+            return scheduler.get_lr()[0]
+        return 0.0
+
 def get_optimizer(
     optimizer_type='BertAdam',
```
Comment (minghui) [inline easynlp/core/optimizers.py:473]: 这个optimizer_type也改成optimizer吧，保持统一

**Review by minghui — APPROVED** [review]
lgtm

**Pull request merged**